### ISSUE TITLE
Multiple problems

### ISSUE THREAD

**Issue by EzellX** [thetrueoneshots/CubeMegaMod#3]
Hi, big fan of the mod but some stuff still isn't working correctly. When disabling the auto gold usage it won't turn back on, Mobs are still invincible, and the weapons that drop from bosses are way too powerful and I am talking in the millions damage and health. I also have yet to see a chest spawn when under water even when the chat says it has. Thanks for all your hard work to keep this game alive and good luck!

**Comment by thetrueoneshots** [comment]
Hey, sorry I noticed it so late! Thanks a lot for the effort of notifying me here on Github. First on your autogoldusage. I have disabled gold usage since v1.3.0. Diving is now just free! 

Also the weapons bosses and creatures drop. Currently they sometimes in some way drop random region weapons. This is normally unusable, but with the CWMod (or known as better progression) it might indeed be sick OP.

As for the chests appearing, I hope to have fixed that in next Sundays update :)

I will look into the dropping of random items by creatures pretty soon as well. 

Thanks again for letting me know, glad you are enjoying the mod and I hope you keep engaged 👍
